src/prefect/runner/runner.py

```diff
@@ -1363,7 +1411,11 @@ async def _submit_run_and_capture_errors(
             return exc
         finally:
             self._release_limit_slot(flow_run.id)
-            self._flow_run_process_map.pop(flow_run.id, None)
+            if TYPE_CHECKING:
+                assert self._flow_run_process_map_lock is not None
```

Comment:
hmm what do you think about a property for that lock?

Reply:
Yup, that's a good call